src/tox/interpreters.py

```diff
@@ -124,57 +124,83 @@ def __str__(self):
 
     @tox.hookimpl
     def tox_get_python_executable(envconfig):
+        py_exe = None
         if envconfig.basepython == "python{}.{}".format(*sys.version_info[0:2]):
-            return sys.executable
-        return py.path.local.sysfind(envconfig.basepython)
+            py_exe = sys.executable
+        if py_exe is None:
+            py_exe = py.path.local.sysfind(envconfig.basepython)
+            if py_exe is None:
+                py_exe = check_python_on_path(version_info(envconfig))
+        return py_exe
 
 
 else:
+    # Exceptions to the usual windows mapping
+    win32map = {"python": sys.executable, "jython": r"c:\jython2.5.1\jython.bat"}
 
     @tox.hookimpl
     def tox_get_python_executable(envconfig):
+        py_exe = None
         if envconfig.basepython == "python{}.{}".format(*sys.version_info[0:2]):
-            return sys.executable
-        p = py.path.local.sysfind(envconfig.basepython)
-        if p:
-            return p
-
-        # Is this a standard PythonX.Y name?
-        m = re.match(r"python(\d)(?:\.(\d))?", envconfig.basepython)
-        groups = [g for g in m.groups() if g] if m else []
-        if m:
-            # The standard names are in predictable places.
-            actual = r"c:\python{}\python.exe".format("".join(groups))
-        else:
-
-            actual = win32map.get(envconfig.basepython, None)
-        if actual:
-            actual = py.path.local(actual)
-            if actual.check():
-                return actual
-        # Use py.exe to determine location - PEP-514 & PEP-397
-        if m:
-            return locate_via_py(*groups)
-
-    # Exceptions to the usual windows mapping
-    win32map = {"python": sys.executable, "jython": r"c:\jython2.5.1\jython.bat"}
+            py_exe = sys.executable
+        if py_exe is None:
+            py_exe = py.path.local.sysfind(envconfig.basepython)
```

Comment:
at least for pre-commit, I've found that this picks up a garbage `python2.7.exe` (from chocolatey) on azure, I switched the find_by_py_launcher to be higher priority

Reply:
how did you switch it, or are you proposing to do this now?

Reply:
I basically just moved it up higher than the "find by exe name" bit: https://github.com/pre-commit/pre-commit/pull/1012/files#diff-235f2434ddb56aaeef64d44558e82943R112

Reply:
did it now 👍